cmd/gen-go-sample/main_test.go

```diff
@@ -160,6 +164,8 @@ func initTestGenerator() *generator {
 			{Name: proto.String("b"), Type: typep(descriptor.FieldDescriptorProto_TYPE_STRING)},
 			{Name: proto.String("e"), TypeName: proto.String(".foo.AType.EType")},
 			{Name: proto.String("f"), Type: typep(descriptor.FieldDescriptorProto_TYPE_STRING), OneofIndex: proto.Int32(0)},
+			{Name: proto.String("file_default"), Type: typep(descriptor.FieldDescriptorProto_TYPE_BYTES)},
+			{Name: proto.String("file_arg"), Type: typep(descriptor.FieldDescriptorProto_TYPE_BYTES)},
```

Comment:
A request field whose contents we happen to read from a file will not necessarily have "file" in the name. I suggest renaming this to "data_bob" for clarity, and making the sample config have `SampleArgumentName: "bob_file"`. That will be a more realistic sample of sample generation usage, I think.

Similarly, make "file_default" into "data_alice". This would then suggest that the variable with contents of the file, when no filename arg is provided, should be named "data_alice" and not "data_alice_contents", whereas for "bob", when the filename arg _is_ provided, it _should_ be "bob_file_contents". WDYT?

If I'm not being clear here, just rename the fields in this proto and the sample config as I suggest and look at the .want file. Do the names make sense?

Reply:
Good point.
Taking one more step further, do you think it makes sense to always use the field name, regardless of whether `SampleArgumentName` is given?

Reply:
As discussed offline: sounds good. We can use a simple algorithm to disambiguate in case of collisions (and we should try to have equivalent algorithms in all our languages). If the user doesn't like how we disambiguate, they'll have to provide a `SampleArgumentName`!